std/algorithm/searching.d

```diff
@@ -3462,6 +3462,181 @@ unittest
 }
 
 /**
+Computes the index of the first occurrence of `range`'s minimum element.
+
+Params:
+    pred = The ordering predicate to use to determine the minimum element.
+    range = The input range to search.
+
+Returns:
+    The index of the first encounter of the minimum element in `range`. If the
+    range is empty, -1 is returned.
+ */
+sizediff_t minIndex(alias pred = "a < b", Range)(Range range)
+    if (isForwardRange!Range && !isInfinite!Range &&
+        is(typeof(binaryFun!pred(range.front, range.front))))
+{
+    if (range.empty) return -1;
+
+    sizediff_t minPos = 0;
+
+    static if (isRandomAccessRange!Range && hasLength!Range)
+    {
+        foreach (i; 1 .. range.length)
+        {
+            if (binaryFun!pred(range[i], range[minPos]))
+            {
+                minPos = i;
+            }
+        }
+    }
+    else
+    {
+        sizediff_t curPos = 0;
+        Unqual!(typeof(range.front)) min = range.front;
+        for (range.popFront(); !range.empty; range.popFront())
+        {
+            ++curPos;
+            if (binaryFun!pred(range.front, min))
+            {
+                min = range.front;
+                minPos = curPos;
+            }
+        }
+    }
+    return minPos;
+}
+
+///
+@safe pure nothrow unittest
+{
+    int[] a = [2, 3, 4, 1, 2, 4, 1, 1, 2];
+    // Minimum is 1 and first occurs in position 3
+    assert(a.minIndex == 3);
+    // Get maximum index with minIndex
+    assert(a.minIndex!"a > b" == 2);
+
+    int[] b;
+    // Range is empty, so return value is -1
+    assert(b.minIndex == -1);
+
+    const int[] c = [2, 5, 4, 1, 2, 3];
+    // Works for const ranges too
+    assert(c.maxIndex == 1);
+    // Lambda
+    assert(c.minIndex!((a, b) => a < b) == 3);
+}
+
+@safe pure unittest
+{
+    // should work with const
+    const(int)[] immArr = [2, 1, 3];
+    assert(immArr.minIndex == 1);
+
+    // should work with immutable
+    immutable(int)[] immArr2 = [2, 1, 3];
+    assert(immArr2.minIndex == 1);
+
+    // with strings
+    assert(["b", "a", "c"].minIndex == 1);
+
+    // infinite range
+    import std.range: cycle;
```

Comment:
[D style](http://dlang.org/dstyle.html) is `import std.range : cycle;`, the motivation here was to have a unified style. This is even enforced by our CircleCi bot, though I just [realized[(https://github.com/dlang/phobos/pull/4955) that the style fix doesn't propagate to PR builds.
Btw you can run the style checks locally with `make -f posix.mak style`